File: offset-manager-remote/src/main/java/service/dto/LastOffsetRecordResponse.java
package service.dto;

import offsetmanager.domain.OffsetRecord;

public record LastOffsetRecordResponse(
  String key,
  long offset
) {
  public static LastOffsetRecordResponse from(OffsetRecord offsetRecord) {
    return new LastOffsetRecordResponse(
      offsetRecord.key(),
      offsetRecord.offset()
    );
  }
}

File: offset-manager-api/src/main/java/offsetmanager/api/v1/dto/LastOffsetRecordResponse.kt
package offsetmanager.api.v1.dto

import offsetmanager.domain.offset.OffsetRecord

data class LastOffsetRecordResponse(
  val key: String,
  val offset: Long
) {
  companion object {
    fun from(offsetRecord: OffsetRecord): LastOffsetRecordResponse {
      return LastOffsetRecordResponse(
        offsetRecord.key().get(),
        offsetRecord.offset()
      )
    }
  }
}
